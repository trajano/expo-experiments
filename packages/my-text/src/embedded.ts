/**
 * This module provides an alternate text renderer that leverages how @expo-google-fonts are used for embedded font scenarios.  It allows a simple migration path from loaded fonts to embedded fonts by rexporting the embedded variants with the same name as the loaded variants.
 */
export { useExpoGoogleFonts } from './useExpoGoogleFonts';
export { MyTextE as MyText } from './MyText';
export { MyTextInputE as MyTextInput } from './MyTextInput';
export { StrongE as Strong, EmE as Em } from './variants';
