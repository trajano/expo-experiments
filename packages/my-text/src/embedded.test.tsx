import { render } from '@testing-library/react-native';
import { createRef } from 'react';
import { Text, TextInput } from 'react-native';
import { mapToEmbeddedFontFamily } from './mapToEmbeddedFontFamily';
import { MyText, Strong, Em, MyTextInput } from './embedded';

describe('MyText via embedded', () => {
  it('should render with embedded font family styles on iOS', () => {
    const { getByText, toJSON } = render(
      <MyText style={{ fontSize: 16 }}>Test Text</MyText>,
    );

    const renderedText = getByText('Test Text');

    // The `mapToEmbeddedFontFamily` is expected to map the font family correctly.
    expect(renderedText.props.style).toMatchObject({
      fontFamily: mapToEmbeddedFontFamily(undefined, undefined, undefined), // Assuming default values are passed
      fontSize: 16,
    });
    expect(toJSON()).toMatchSnapshot();
  });

  it('should handle variants', () => {
    const { getByText, toJSON } = render(
      <MyText style={{ fontSize: 16 }}>
        <Strong>
          <Em>Test Text</Em>
        </Strong>
      </MyText>,
    );

    const renderedText = getByText('Test Text');

    // The `mapToEmbeddedFontFamily` is expected to map the font family correctly.
    // font size is not transfered because it is not managed by MyText
    expect(renderedText.props.style).toMatchObject({
      fontFamily: mapToEmbeddedFontFamily(undefined, undefined, undefined), // Assuming default values are passed
      fontWeight: 'bold',
      fontStyle: 'italic',
    });
    expect(toJSON()).toMatchSnapshot();
  });

  it('should forward the ref to the underlying Text component', () => {
    const ref = createRef<Text>();

    const { getByText } = render(<MyText ref={ref}>Test Text</MyText>);

    const renderedText = getByText('Test Text');

    // Ensure that the ref is correctly assigned to the underlying Text component
    expect(ref.current).toBeInstanceOf(Text);
    expect(ref.current.props).toEqual(renderedText.props);
  });

  it('should forward the ref to the underlying Text component', () => {
    const ref = createRef<TextInput>();

    const { getByTestId } = render(<MyTextInput testID="input" ref={ref} />);

    const inputField = getByTestId('input');

    // Ensure that the ref is correctly assigned to the underlying Text component
    expect(ref.current).toBeInstanceOf(TextInput);
    expect(ref.current.props).toEqual(inputField.props);
  });
});
