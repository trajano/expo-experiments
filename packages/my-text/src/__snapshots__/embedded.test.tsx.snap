// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyText via embedded should handle variants 1`] = `
<Text
  style={
    {
      "fontFamily": undefined,
      "fontSize": 16,
      "fontStyle": undefined,
      "fontWeight": undefined,
      "lineHeight": undefined,
    }
  }
>
  <Text
    style={
      {
        "fontFamily": undefined,
        "fontStyle": undefined,
        "fontWeight": "bold",
        "lineHeight": undefined,
      }
    }
  >
    <Text
      style={
        {
          "fontFamily": undefined,
          "fontStyle": "italic",
          "fontWeight": "bold",
          "lineHeight": undefined,
        }
      }
    >
      Test Text
    </Text>
  </Text>
</Text>
`;

exports[`MyText via embedded should render with embedded font family styles on iOS 1`] = `
<Text
  style={
    {
      "fontFamily": undefined,
      "fontSize": 16,
      "fontStyle": undefined,
      "fontWeight": undefined,
      "lineHeight": undefined,
    }
  }
>
  Test Text
</Text>
`;
